solutions/usaco-808.mdx

```diff
@@ -66,4 +73,144 @@ for i in range(n):
 print(ans)
 ```
 </PySection>
+<CPPSection>
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+	
+```cpp
+#include <iostream>
+#include <fstream>
+using namespace std;
+int N, cows[100], cowsPassingTo[100];                                                                      
+// Which cow does a given cow pass to?
+int passToCow(int cow) {
+	int leftCows = -1;
+	int rightCows = -1;
+	int leftRange = 1000;
+	int rightRange = 1000;
+	// For each cow, find closest cows to it on the left and right
+	for (int i = 0; i < N; i++) {
+		if (cows[i] < cows[cow] && cows[cow] - cows[i] < leftRange) {
+			leftCows = i;
+			leftRange = cows[cow] - cows[i];
+		}
+		if (cows[i] > cows[cow] && cows[i] - cows[cow] < rightRange) {
+			rightCows = i;
+			rightRange = cows[i] - cows[cow];
+		}
+	}
+	// Return number of cows passing to the given cow
+	if (leftRange <= rightRange) {
+		return leftCows;
+	}		
+	return rightCows;
+}
+int main(void)
+{
+	ifstream fin ("hoofball.in");
+	ofstream fout ("hoofball.out");
+  	fin >> N;
+	int ballsNeeded = 0;
+  	for (int i=0; i<N; i++) {
+		fin >> cows[i];
+	}
+  	for (int i=0; i<N; i++) {
+		cowsPassingTo[passToCow(i)]++;
+	}
+  	for (int i=0; i<N; i++) {
```

Comment:
```suggestion
  	for (int i = 0; I < N; i++) {
```
normally this is optional but you have spaces everywhere else

Reply:
brad i think you mistyped `I`

Reply:
fixed, also you can just directly commit these small changes